docs/scos/dev/guidelines/monitorable-process-guidelines.md

```diff
@@ -0,0 +1,86 @@
+can be considered a helpful addition.
+* Avoid logging in performance critical places. In case there is a performance critical process or an unusually large cycle as a sub-process
+element, you should make a careful decision between performance vs traceability.
+* Audit trail (including authentication & authorization): it is recommended to log the audit trails of important entity changes but it should 
+not be sent to the regular log system since the typical data here consists mainly of sensitive data.
+
+### Log levels
+The following log levels MUST be utilised appropriately during application workflows to allow easy track down particular problems / investigations 
+in the System. A wrong category selection MAY de-rail a fellow inspector.
+
+
+| Level | Description |
+|-------|-------------|
+| Debug | Provides detailed data, which is mostly used for debugging. These logs allow to check variable values and/or error stacks. |
+| Info | Interesting events. |
+| Notice | Uncommon events. |
+| Warning | Describes events that are less destructive than errors. They usually do not result in any reduction of the program's functionality or its full failure. Undesirable things that are not necessarily wrong. |
+| Error | Identifies error events that may still let the software run, but with restricted capabilities in the impacted routes. |
+| Critical | Identifies extremely serious error events that are likely to cause the program to abort. Typically, this leads to catastrophic failures. |
+| Alert | Action must be taken immediately. |
+| Emergency | An urgent alert. |
+
+### Log entry format
+A log entry MUST always answer the following items and follow the below describes structure
+* **who** caused the event
+* **when** exactly did the event happened
+* **where** did the event took place (context, component, application, etc.)
+* **what** has happened and/or why has it happened
+* **result** of event (exception, success details, information details)
+
+**TODO: add code example blocks from https://spryker.atlassian.net/wiki/spaces/ALO/pages/3388014860/Monitorable+process+guidelines**
+
+## Metric generation
+Applies to all application components under ACI scoped processes.
+
+Each metric represents a condition of some system attribute. You can have a lot of them and correlate them with each other.
+
+* Every service / component should define and generate project appropriate metrics for key processes (with basic dimension of the outcome 
+of the operation (success, failure) and duration) to enable us tracking such events and react when they reach undesired scores.
+   * Start and end of a process must be sent to enable us tracking and tuning infrastructure.
+   * Communication durations with remote services must be sent to us to enable us understanding that the local process is delayed for a good reason.
+* Critical metrics together with their threshold vision must be highlighted in the Operational guideline to enable us setting up monitoring system.
```

Comment:
> highlighting critical metrics in Operational guideline
Correct me if I'm wrong, but sounds like there needs to be a process, where our OPS team seats with the project team to define those metrics to setup the right monitoring? Sounds nice, but not sure if we wanna do that with every customer..?

Reply:
To be clarified

Reply:
correct